Added api that handles events

battleangel217 opened the pull request.

battleangel217 committed:
Added api that handles events

mikelneonedwin committed:
Merge branch 'main' into main

mikelneonedwin commented on api.py:
```diff
@@ -50,6 +50,105 @@ def register_for_event(event_id):
+
+@api.route("/events", methods=["POST", "GET"])
+def addevent():
+    if request.method == "POST":
+
+        conn = get_db_connection()
+        cursor = conn.cursor()
+
+        user_id = request.cookies.get("user_id")
+        name = request.form.get("title")
+        description = request.form.get("description")
+        date = request.form.get("date")
+        location = request.form.get("location")
+        price = request.form.get("price")        
+
+        cursor.execute("INSERT INTO events (creator_id, title, description, date, location, price) VALUES(?, ?, ?, ?, ?, ?)", (user_id, name, description, date, location, price))
+
+        conn.commit()
+        conn.close()
+
+        return jsonify({"message":"Event added successfully"}), 201
```
Before creating the event, you're to verify that the user actually exists

mikelneonedwin commented on api.py:
```diff
@@ -50,6 +50,105 @@ def register_for_event(event_id):
+        if event:
+            return jsonify(dict(event))
+        else:
+            return jsonify({"error":"Event not found"})
```
When the data doesn't exist, we return a 404

mikelneonedwin commented on api.py:
```diff
@@ -50,6 +50,105 @@ def register_for_event(event_id):
+
+
+        #Get the events created by the user, so other users can't edit another user's event's
+        cursor.execute("SELECT * FROM events WHERE id = ? AND creator_id = ?", (eventid, id))
```
This is nice but normally this is what we do
- Check that the event exists
- Authenticate the user
- Confirm that the user is authorized to update this event (i.e., they are the owner of the event)

Yes, it is not too necessary for this project, but I feel like I should let you know

mikelneonedwin commented on api.py:
```diff
@@ -50,6 +50,105 @@ def register_for_event(event_id):
+
+        if not event:
+            conn.close()
+            return jsonify({"error":"Event not found"})
```
Once again, we return a 404 to let them know the event does not exist

mikelneonedwin commented on api.py:
```diff
@@ -50,6 +50,105 @@ def register_for_event(event_id):
+
+        if not event:
+            conn.close()
+            return jsonify({"error":"Event not found"})
```
404 once again

mikelneonedwin commented on api.py:
```diff
@@ -50,6 +50,105 @@ def register_for_event(event_id):
+        id = request.cookies.get("user_id")
+
+
+        cursor.execute("SELECT * FROM events WHERE id = ? AND creator_id = ?", (eventid, id,))
```
Remember the comment I made up there about updating events?

mikelneonedwin commented on api.py:
```diff
@@ -50,6 +50,105 @@ def register_for_event(event_id):
+        conn.commit()
+        conn.close()
+
+        return jsonify({"message": "Event deleted successfully"}), 200
```
Our API schema specified that you return a 204 though

mikelneonedwin commented on api.py:
```diff
@@ -50,6 +50,105 @@ def register_for_event(event_id):
+        conn.close()
+
+        return jsonify({"message": "Event deleted successfully"}), 200
+=======
```
This will cause syntax errors, do well to checkout the this [Tutorial](https://www.youtube.com/playlist?list=PL4cUxeGkcC9goXbgTDQ0n_4TBzOO0ocPR) it'l help you use Git, GitHub and VS Code to contribute better.

mikelneonedwin reviewed (commented):
Just identified some issues with your PR

mikelneonedwin committed:
Update api.py

Fixed issues with how user id is being used (string instead of integer), corrected a few status codes, and some house keeping
